### PR TITLE
Opacities action

### LINKED ISSUE #311: Get the transparency tool to work on a range
Currently we just have two levels - opaque (opacity level at 1) and transparent (opacity level at 0.7)
transparency

Relevant lines of code are:

1) an `opacity` option for the 0.7 value
https://github.com/publiclab/Leaflet.DistortableImage/blob/c2bf3def17149735ab5790ae5f4a09fa9be513eb/src/edit/DistortableImage.Edit.js#L3-L20

2) The `_toggleTransparency` action: 
https://github.com/publiclab/Leaflet.DistortableImage/blob/c2bf3def17149735ab5790ae5f4a09fa9be513eb/src/edit/DistortableImage.Edit.js#L294-L303

3) The corresponding toolbar icon code:
https://github.com/publiclab/Leaflet.DistortableImage/blob/c2bf3def17149735ab5790ae5f4a09fa9be513eb/src/edit/DistortableImage.EditToolbar.js#L12-L26

It would be great to get the transparency tool to work on a range – going from opaque to increasingly more transparent.

### PR DESCRIPTION
**Pull request by themacboy**
Fixes #311

First of all, sorry for my poor English and weak coding skills (I do it just for fun!)

I was working on a new action to set custom and variable opacity.

Please check the video for more details:

[Opacities Action](https://www.youtube.com/watch?v=E5hELJwruzY)

You can play it:

[Opacities Action example page](https://themacboy.github.io/Leaflet.DistortableImage/examples/)

### CODE REVIEW

**Comment by themacboy** [comment]
Does anyone like this code? haha

I feel lonely! ;)

**Comment by vanithaak** [comment]
> Does anyone like this code? haha
> 
> I feel lonely! ;)

Hey! I like your code. Nice work 👏

**Comment by 7malikk** [comment]
@themacboy Its great!!
I went through it before you closed the PR the first time.
I quite like the idea
@jywarren should check it out eventually 

PS. glad you reopened it 🙌🏾🙌🏾

**Comment by themacboy** [comment]
I'm currently using CSSStyleSheet to define my css overrides... but that's too modern and not all browsers have a correct implementation.

Then I'm not sure about changing it to the traditional <style> tag.

**Comment by alenigrelli** [comment]
can you merge in the project?. Thanks!

File: examples/index.html [hunk 1]
```diff
@@ -6,12 +6,12 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no"/>
     <meta http-equiv="x-ua-compatible" content="ie=edge"/>
 
-    <link rel="stylesheet" href="../node_modules/leaflet/dist/leaflet.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.9.2/leaflet.min.css">
```
Comment (jywarren) [inline examples/index.html:9]: Here, i'm not against pegging a specific version via CDN, but was this necessary to make this work? Just curious if we could remove this to simplify the PR. Thanks!

File: examples/index.html [hunk 2]
```diff
@@ -6,12 +6,12 @@
...
     <link rel="stylesheet" href="../dist/vendor.css">
     <link rel="stylesheet" href="../dist/leaflet.distortableimage.css" media="screen" title="no title">
     <link rel="stylesheet" href="https://unpkg.com/leaflet-control-geocoder/dist/Control.Geocoder.css" />
 
-    <script src="../node_modules/leaflet/dist/leaflet.js" type="text/javascript" charset="utf-8"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.9.2/leaflet.js" type="text/javascript" charset="utf-8"></script>
```
Comment (jywarren) [inline examples/index.html:14]: Same here. Thanks!

**Review by jywarren — COMMENTED** [review]
Hi, sorry for my slow review here, been busy! Much appreciated, this looks really cool, and I love the new icon! 

Just to clarify, is this on by default in one of the examples? Also, a Q below about the CDN leaflet. We could upgrade Leaflet itself if that's preferable to switching to a CDN.

**Comment by jywarren** [comment]
Also, THANK YOU for such great video documentation, live example, tidy code, and adjustments to the tests!

File: examples/index.html (hunk 1 above)
Comment (themacboy) [reply examples/index.html]: Hello ! Im alive !!!

Thx for your all yout comments !

The reason for moving to CDN was due to Github Pages service is failing when trying to load local files (missing files).

image

Then I couldn't test my changes on the web, lol, that's the only reason for this change.

Comment (themacboy) [reply examples/index.html]: If needed I can set local method before the final merge ;)

Comment (jywarren) [reply examples/index.html]: Oh that would be great to revert this back to the local one. We use that in our gh-pages demo. Thank you!

File: examples/index.html (hunk 2 above)
Comment (jywarren) [reply examples/index.html]: And here too!

**Comment by jywarren** [comment]
And just one final question - is this on by default in one of the examples? Thanks!

**Comment by jywarren** [comment]
(i see you adjusted the CDNs so otherwise approved! Thanks!)

**Comment by themacboy** [comment]
@jywarren update with updating new commits.

**Comment by jywarren** [comment]
Thanks!! Just wanted to know that last question -- is this on by default in one of the examples? Thanks!

**Comment by themacboy** [comment]
> Thanks!! Just wanted to know that last question -- is this on by default in one of the examples? Thanks!

yes ! I think it is.

**Review by jywarren — APPROVED** [review]

**Comment by jywarren** [comment]
Wonderful, thank you!!!!!

**Pull request merged**